### PR TITLE
Improvements

### LINKED ISSUE #81: Support trans.diat and trans.semi
It seems the plugin does not generate the trans.diat and trans.semi attributes

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR brings
* MIDI velocity to notes
* adds transposition attributes to `staffDef`

closes #81

### CODE REVIEW

File: src/ExportConverters.mss [hunk 1]
```diff
@@ -1047,49 +1047,41 @@ function ConvertEndingValues (styleid) {
...
         }
         case ('open')
         {
-            type = 'open';
+            lendsym = 'none';
         }
         case ('closed')
         {
-            type = 'closed';
+
```
Comment (th-we) [inline src/ExportConverters.mss:1084]: Shouldn't we add `lendsym='angledown'` here? In fact, `angledown` should be added to all the styles, except those that have a traling `.open` in their style ID – those should have `lendsym='none'`.

File: src/ExportConverters.mss [hunk 2]
```diff
@@ -1047,49 +1047,41 @@ function ConvertEndingValues (styleid) {
...
-            label = '1.';
-            type = 'closed';
+            num = '1.';
         }
         case ('1st_n_2nd')
         {
-            num = 1;
-            label = '1. 2.';
-            type = 'closed';
+            num = '1. 2.';
```
[the remark this answers is not in the document]
Comment (rettinghaus) [reply src/ExportConverters.mss:1060]: Ah, yes the space has got to go. But the `@n` here is (in MEI4) of type `att.nNumberLike`, so this is *exactly* what this is for. And the nicer rendition in Verovio is of course welcome.

File: src/ExportConverters.mss (hunk 1 above)
Comment (rettinghaus) [reply src/ExportConverters.mss:1084]: I would guess, that the default style for ending brackets is "closed" and so open brackets need explicitly a `@lendsym="none"`.

File: src/ExportConverters.mss (hunk 2 above)
[the remark this answers is not in the document]
Comment (annplaksin) [reply src/ExportConverters.mss:1060]: But in MEI4 spaces in `@n` are forbidden. Its content is `data.WORD` which cannot contain whitespaces.

[the remark this answers is not in the document]
Comment (ahankinson) [reply src/ExportConverters.mss:1060]: Does Verovio render `@label`? That's probably more appropriate here.

[the remark this answers is not in the document]
Comment (rettinghaus) [reply src/ExportConverters.mss]: @ahankinson No, Verovio does not render `@label`. Furthermore `@label` is not even close to being appropriate as it "Should not be used to record document content." See https://music-encoding.org/guidelines/v4/attribute-classes/att.labelled.html

[the remark this answers is not in the document]
Comment (ahankinson) [reply src/ExportConverters.mss]: "Not even close" is a bit much (you **can** make the same point a bit less aggressively), but ok.

[the remark this answers is not in the document]
Comment (rettinghaus) [reply src/ExportConverters.mss]: @ahankinson sorry, didn't meant to be rude.

**Comment by annplaksin** [comment]
@rettinghaus Would you please add some example files in the `test/sibmeiTestSibs`? This would make it easier to write tests... I think we should try to have tests for the changes we made before we prepare the next release. (It would as well make it easier to review)

File: src/ExportConverters.mss (hunk 2 above)
[the remark this answers is not in the document]
Comment (rettinghaus) [reply src/ExportConverters.mss]: as we can't agree to the changes on endings I removed this commit.

**Comment by th-we** [comment]
@annplaksin Using arrow functions in mocha tests [is discouraged](https://mochajs.org/#arrow-functions). I created [another pull request](https://github.com/rettinghaus/sibmei/pull/3) on @rettinghaus' pull request to change that.

File: src/ExportGenerators.mss [hunk 3]
```diff
@@ -1027,6 +1027,7 @@ function GenerateNote (nobj) {
 
     ntinfo = ConvertDiatonicPitch(nobj.DiatonicPitch);
     pnum = nobj.Pitch;
+    vel = nobj.OriginalVelocity;
```
Comment (th-we) [inline src/ExportGenerators.mss:1031]: Unless there's a reason to declare this as a variable I'd suggest just doing `libmei.AddAttribute(n, 'vel', nobj.OriginalVelocity);` – easier to see where the value comes from.

**Comment by annplaksin** [comment]
@th-we I did it like it was done in the previous tests. If this is now discouraged, we maybe should change that for every test.

**Comment by th-we** [comment]
@annplaksin Oh, good you noticed!  I'll have to change that.  (Only became aware of that recommendation after I wrote the initial tests.)

**Comment by annplaksin** [comment]
@th-we Thanks for mentioning it. I wrote all my tests by adopting what you did... ;-)

**Pull request merged**